### PR TITLE
Add basic unittest debugging guide

### PR DESCRIPTION
**Pull request by diabelko**

### CODE REVIEW

File: source/user/debugging.rst [hunk 1]
```diff
@@ -13,4 +13,18 @@ There are two types of review dependencies: implicit and explicit.
...
+Logs
+````
+You should probably start your debugging with checking what tests failed - you can do this in ``scons_test-FAIL.log.gz``. You can look for the tests and their results in ``job-output.txt.gz`` looking for the test name. 
+
+If this doesn't help you, try looking for ``scons: *`` to see all errors raised by SCons - most often failures will be precedented by Python stacktrace.
+
+
+Flakiness
+`````````
+Sometimes it's just test flakiness and not much you can do. To run the tests again, type ``recheck`` in a comment in your review.
```
Comment (jluk-codi) [inline source/user/debugging.rst]: Add some information about possible further steps:
- letting know about a broken/flaky test on the #ci channel (tungstenfabric slack)
- contacting the test author, if known

File: source/user/debugging.rst [hunk 2]
```diff
@@ -13,4 +13,18 @@ There are two types of review dependencies: implicit and explicit.
...
+**Explicit dependency** is a dependency specified in git commit message by ``Depends-On: $Change-ID``
+
+Unittest failures
+-----------------
+
+Logs
+````
+You should probably start your debugging with checking what tests failed - you can do this in ``scons_test-FAIL.log.gz``. You can look for the tests and their results in ``job-output.txt.gz`` looking for the test name. 
+
+If this doesn't help you, try looking for ``scons: *`` to see all errors raised by SCons - most often failures will be precedented by Python stacktrace.
```
Comment (jluk-codi) [inline source/user/debugging.rst]: "try searching for the `scons: *` string"?

File: source/user/debugging.rst [hunk 3]
```diff
@@ -13,4 +13,18 @@ There are two types of review dependencies: implicit and explicit.
...
 
-**Explicit dependency** is a dependency specified in git commit message by ``Depends-On: $Change-ID``.
+**Explicit dependency** is a dependency specified in git commit message by ``Depends-On: $Change-ID``
+
+Unittest failures
+-----------------
+
+Logs
+````
+You should probably start your debugging with checking what tests failed - you can do this in ``scons_test-FAIL.log.gz``. You can look for the tests and their results in ``job-output.txt.gz`` looking for the test name.
```
Comment (jluk-codi) [inline source/user/debugging.rst]: "which tests"?

**Review by jluk-codi — CHANGES_REQUESTED** [review]

**Review by jluk-codi — DISMISSED** [review]

File: source/user/debugging.rst [hunk 4]
```diff
@@ -13,4 +13,20 @@ There are two types of review dependencies: implicit and explicit.
...
+Logs
+````
+You should probably start your debugging with checking what tests failed - you can do this in ``scons_test-FAIL.log.gz``. You can search for the tests and their results in ``job-output.txt.gz`` using the test name. 
+
+If this doesn't help you, try searching for ``scons: *`` to see all errors raised by SCons - most often failures will be preceded by Python stacktrace.
+
+
+Flakiness
+`````````
+Sometimes it's just test flakiness and not much you can do. To run the tests again, type ``recheck`` in a comment in your review.
```
Comment (wurbanski) [inline source/user/debugging.rst]: > Sometimes it's just test flakiness **and not much you can do**.

and there is not much you can do? and you can't do much?

**Review by wurbanski — DISMISSED** [review]

**Review by jluk-codi — APPROVED** [review]

**Pull request merged**